src/server/UdpServerSockEP.cpp

```diff
@@ -44,6 +45,13 @@ UdpServerSockEP::UdpServerSockEP(std::string ipaddr, int port, std::function<voi
 		return;
 	}
 
+	// Join multicast group if given. If interfaceAddr is not given, use INADDR_ANY and the system will choose one.
+	if (multicastAddr != "")
+	{
+		joinMulticastGroup(interfaceAddr, multicastAddr);
```

Comment:
This should return a success indicator, and if it failed to join the multicast group, the socket should be marked invalid.